src/components/DateInput/DateInput.tsx

```diff
@@ -0,0 +1,343 @@
+      setDate(stateDateObj);
+    }
+    if (isRange && rangeConfig) {
+      setRange(rangeConfig);
+    }
+  }, []);
+
+  const handleDateInput = (
+    event: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>,
+    dateKey: string,
+    index: number
+  ) => {
+    let stateDateObj: any = { ...date };
+    event.target.value = event.target.value.replace(/[^0-9]+/g, "");
+    // if the given input has a length on 2 charcters and its not the last input field
+    if (event.target.value.length == 2 && index != 2) {
+      // get the next document and focus
+      let nextInput = document.getElementById(`date-input-${index + 1}-${id}`);
+      nextInput?.focus();
+    }
+
+    if (index == 2 && event.target.value.length > 4) {
+      return;
+    }
+    // set the data in the state
+    stateDateObj[dateKey] = event.target.value;
+    setDate(stateDateObj);
+    // if all inputs are filled then execute the local date validation function
+    if (stateDateObj.mm && stateDateObj.dd && stateDateObj.yyyy) {
+      defaultErrorValidation(stateDateObj);
+    }
+    getDateValue?.(
+      new Date(`${stateDateObj.mm}/${stateDateObj.dd}/${stateDateObj.yyyy}`)
+    );
+  };
+
+  // Function to handle cancel button click
+  const handleCancelClick = () => {
+    let stateDateObj = { ...date };
+    stateDateObj.mm = "";
+    stateDateObj.dd = "";
+    stateDateObj.yyyy = "";
+    setDate(stateDateObj);
+    // empty state value and error text
+    setDateError("");
+    getDateValue?.("");
+  };
+
+  // a default date validation that peforms basic checks
+  const defaultErrorValidation = (stateDateObj: any) => {
+    if (stateDateObj.mm && (stateDateObj.mm == 0 || stateDateObj.mm > 12)) {
+      setDateError("Invalid Date");
+      return;
+    }
+    if (
+      (stateDateObj.yyyy && stateDateObj.yyyy.length < 4) ||
+      ["0000", "9999"].includes(stateDateObj.yyyy)
+    ) {
+      setDateError("Invalid Date");
+      return;
+    }
+    let availableMonths = daysInMonth(stateDateObj.mm, stateDateObj.yyyy);
+    if (
+      stateDateObj.dd &&
+      (stateDateObj.dd == 0 || stateDateObj.dd > availableMonths)
+    ) {
+      setDateError("Invalid Date");
+      return;
+    }
+    // if no errors then set the error state as empty
+    setDateError("");
+  };
+
+  // Function to get number of days available in a mont
+  function daysInMonth(month: number, year: number) {
+    return new Date(year, month, 0).getDate();
+  }
+
+  function handleBackSpace(currentIndex: number, event: any) {
+    // if there is a field before the current index then delete and focus on that index
+    if (currentIndex >= 0 && !event.target.value) {
+      let prevField = document.getElementById(
+        `date-input-${currentIndex - 1}-${id}`
+      );
+      prevField?.focus();
+    }
+    defaultErrorValidation(date);
+  }
+
+  //function to handle backspace event
+  function handleKeyDown(event: any, currentIndex: number) {
```

Comment:
pls check if we can use use event type `event: React.KeyboardEvent<HTMLInputElement>` for type safety

Reply:
Changes made 👍